tests/dualtor/test_ipinip.py

```diff
@@ -0,0 +1,145 @@
+    _, server_ipv4 = test_interface
+    config_facts = tor.get_running_config_facts()
+    try:
+        peer_ipv4_address = [_["address_ipv4"] for _ in config_facts["PEER_SWITCH"].values()][0]
+    except IndexError:
+        raise ValueError("Failed to get peer ToR address from CONFIG_DB")
+
+    tor_ipv4_address = [_ for _ in config_facts["LOOPBACK_INTERFACE"]["Loopback0"]
+                        if is_ipv4_address(_.split("/")[0])][0]
+    tor_ipv4_address = tor_ipv4_address.split("/")[0]
+
+    inner_dscp = random.choice(range(0, 33))
+    inner_ttl = random.choice(range(3, 65))
+    inner_packet = testutils.simple_ip_packet(
+        ip_src="1.1.1.1",
+        ip_dst=server_ipv4,
+        ip_dscp=inner_dscp,
+        ip_ttl=inner_ttl
+    )[IP]
+    packet = testutils.simple_ipv4ip_packet(
+        eth_dst=tor.facts["router_mac"],
+        eth_src=ptfadapter.dataplane.get_mac(0, 0),
+        ip_src=peer_ipv4_address,
+        ip_dst=tor_ipv4_address,
+        ip_dscp=inner_dscp,
+        ip_ttl=255,
+        inner_frame=inner_packet
+    )
+    logging.info("the encapsulated packet to send:\n%s", tunnel_traffic_monitor._dump_show_str(packet))
+    return packet
+
+
+def get_ptf_server_intf_index(tor, tbinfo, iface):
+    """Get the index of ptf ToR-facing interface on ptf."""
+    mg_facts = tor.get_extended_minigraph_facts(tbinfo)
+    return mg_facts["minigraph_ptf_indices"][iface]
+
+
+def build_expected_packet_to_server(encapsulated_packet):
+    """Build packet expected to be received by server from the tunnel packet."""
+    inner_packet = encapsulated_packet[IP].payload[IP].copy()
+    # use dummy mac address that will be ignored in mask
+    inner_packet = Ether(src="aa:bb:cc:dd:ee:ff", dst="aa:bb:cc:dd:ee:ff") / inner_packet
+    exp_pkt = mask.Mask(inner_packet)
+    exp_pkt.set_do_not_care_scapy(Ether, "dst")
+    exp_pkt.set_do_not_care_scapy(Ether, "src")
+    exp_pkt.set_do_not_care_scapy(IP, "tos")
+    exp_pkt.set_do_not_care_scapy(IP, "ttl")
+    exp_pkt.set_do_not_care_scapy(IP, "chksum")
+    return exp_pkt
+
+
+def test_dualtor_ipinip_active_tor(
+    apply_mock_dual_tor_tables,
+    apply_mock_dual_tor_kernel_configs,
+    apply_active_state_to_orchagent,
+    build_encapsulated_packet, test_interface, ptfadapter,
+    tbinfo, tor, tunnel_traffic_monitor
+):
+    encapsulated_packet = build_encapsulated_packet
+    iface, _ = test_interface
+
+    exp_ptf_port_index = get_ptf_server_intf_index(tor, tbinfo, iface)
+    exp_pkt = build_expected_packet_to_server(encapsulated_packet)
+
+    ptfadapter.dataplane.flush()
+    ptf_t1_intf = random.choice(get_t1_ptf_ports(tor, tbinfo))
+    logging.info("send encapsulated packet from ptf t1 interface %s", ptf_t1_intf)
+    testutils.send(ptfadapter, int(ptf_t1_intf.strip("eth")), encapsulated_packet, count=1)
+    _, rec_pkt = testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[exp_ptf_port_index])
+    rec_pkt = Ether(rec_pkt)
```

Comment:
I suggest adding a timeout for ```verify_packet_any_port``` instead of using the default value(```None```).

Reply:
`ptf` has default timeout as 2 seconds, which should be sufficient in this scenario.